codemon/codemon.py

```diff
@@ -4,42 +4,76 @@
-from codemon.CodemonMeta import template_cpp,get_filename,get_practice_files
+from codemon.CodemonInit import init, initjava
+from codemon.CodemonMeta import template_cpp, get_filename, get_practice_files, template_java, get_filename_java, get_practice_files_java
+
 
 def main():
   if len(sys.argv) < 2:
     showHelp()
 
   else:
-    countArg = 0
+    countArg = 0;
+    isCppFile = False
+    isJavaFile = False
+    isSingleFile = False
+    toInit = False
+    toListen = False
+    toPractice = False
+    targetFile = ""
+    # get all arguments and flags
     for arg in sys.argv:
-      countArg+=1
-
+      countArg+=1;
       if arg == "init":
-        if sys.argv[countArg] == '-n':
-          file = sys.argv[countArg+1]
-          path = '.'
-          f = open(path + '/' + file + '.cpp',"w+")
-          template = template_cpp()
-          f.write(template)
-          f.close()
-          print(colored.yellow("Created "+file+'.cpp'))
-          break
-
-        else:
-          contestName = sys.argv[countArg]
-          fileNames = get_filename()
-          init(contestName, fileNames)
-
+        toInit = True
+      elif arg == "-cpp":
+        isCppFile = True
+      elif arg == "-java":
+        isJavaFile = True
+      elif arg == "-n":
+        isSingleFile = True
       elif arg == "listen":
-        listen()
-
+        isListen = True
       elif arg == "practice":
-        contestName = sys.argv[countArg]
+        toPractice = True
+      else:
+        targetFile = arg
+
+    # based on booleans perform action
+    if toListen == True:
+      listen()
+
+    elif toInit == True:
+      if isJavaFile == True and isSingleFile == False:
+        name = input("Enter Contest Name: ")
+        files = get_filename_java()
+        initjava(name, files)
+
+      elif isJavaFile == True and isSingleFile == True:
+        # extension & path passed directly here
+        fileName = input("Enter filename: ")
+        dirName = input("Enter directory: ")
+        file = [fileName + '.java']
+        initjava(dirName, file)
+      
+      elif isCppFile == True and isSingleFile == False:
+        name = input("Enter Contest Name: ")
+        files = get_filename()
+        init(name, files)
+
+      elif isCppFile == True and isSingleFile == True:
+        # extension & path passed directly here
+        fileName = input("Enter filename: ")
+        dirName = input("Enter directory: ")
+        file = [fileName + '.cpp']
+        init(dirName, file)
+
+      elif isCppFile == True and toPractice == True:
+        name = input("Enter Contest Name: ")
         practiceFiles = get_practice_files()
-        init(contestName, practiceFiles)
+        init(name, practiceFiles)
 
-      elif arg == "--help":
-        showHelp()
-        break
+      elif isJavaFile == True and toPractice == True:
+        name = input("Enter Contest Name: ")
+        practiceFiles = get_practice_files_java()
+        initjava(name, practiceFiles)
```

Comment:
It ends here, do I have to include the listen fn @ankingcodes ?? And what else is needed?